xbmc/windowing/wayland/WinSystemWayland.h

```diff
@@ -108,10 +111,12 @@ class CWinSystemWayland : public CWinSystemBase, IInputHandler, IWindowDecoratio
   {
     return m_surface;
   }
+  wayland::compositor_t GetCompositor() { return m_compositor; }
```

Comment:
Isn't `CWinSystemWaylandWebOS` deriving from`CWinSystemWayland` ? Why does the base class need a public getter method for the compositor?
Seems a bit wrong to expose internal members this way.

Reply:
Agreed we can register another private compositor.

Reply:
the compositor_t is just an interface class without any state that WinSystemWayland requests anyway, so I think it's fine to export it as a courtesy. it's not really "private" in that sense.

Reply:
I've already changed it to bind another compositor. You think it's better as it was before?